google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb

```diff
@@ -183,21 +191,54 @@ def self.to_query_param value
                 type: struct_param.parameter_type
               ), struct_param.parameter_value]
             end
+            struct_values = Hash[struct_pairs.map do |type, value|
+              [type.name, value]
+            end]
 
             return Google::Apis::BigqueryV2::QueryParameter.new(
               parameter_type: Google::Apis::BigqueryV2::QueryParameterType.new(
                 type: "STRUCT",
                 struct_types: struct_pairs.map(&:first)
               ),
               parameter_value: Google::Apis::BigqueryV2::QueryParameterValue.new(
-                struct_values: struct_pairs.map(&:last)
+                struct_values: struct_values
               )
             )
           else
             fail "A query parameter of type #{value.class} is not supported."
           end
         end
 
+        ##
+        # @private
+        def self.to_json_rows rows
+          rows.map do |row|
+            Hash[row.map { |k, v| [k.to_s, to_json_value(v)] }]
+          end
+        end
+        ##
+        # @private
+        def self.to_json_value value
+          if DateTime === value
```

Comment:
Yeah, a case statement uses `===`. The IO support makes it hard to use a case statement though, so we have this syntax. We have similar code in Datastore and Spanner, for the same reason.